### ISSUE TITLE
Snippet plug-ins will not work properly

### ISSUE THREAD

**Issue by tamago324** [dstein64/nvim-scrollview#28]
When the scrollbar is displayed, the placeholder string becomes `1G` when expanded with [neosnippet](https://github.com/Shougo/neosnippet.vim).
### System info

- `nvim --version`: `NVIM v0.5.0-dev+1133-g0869cbd55`
- Operating system/version: `Manjaro Linux 20.2.1`

### Steps to reproduce using `nvim -u min_vimrc -i NONE`

1. `$ nvim -u min_vimrc -i NONE`
2. `:normal! 15o`
3. `:resize 10` (display scrollbar)
4. `:normal! gg`
5. `:set ft=vim`
6. input `func`
7. press `<C-k>` (execute `<Plug>(neosnippet_jump_or_expand)`)

```vim
" min_vimrc
set encoding=utf-8

filetype plugin indent on
if has('vim_starting')
  let s:pluin_manager_dir='~/.config/nvim/.plugged/vim-plug'
  execute 'set runtimepath+=' . s:pluin_manager_dir
endif
call plug#begin('~/.config/nvim/.plugged')
Plug 'dstein64/nvim-scrollview'
Plug 'Shougo/neosnippet.vim'
Plug 'Shougo/neosnippet-snippets'
call plug#end()

set nobackup
set nowritebackup
set noswapfile
set updatecount=0
set backspace=indent,eol,start
set number
language messages en_US.utf8

lua << EOF
vim.g.scrollview_winblend = 80
vim.g.scrollview_column = 1
vim.g.scrollview_current_only = 1
EOF

imap <C-k> <Plug>(neosnippet_jump_or_expand)
smap <C-k> <Plug>(neosnippet_jump_or_expand)
```

https://user-images.githubusercontent.com/16581287/111422428-09f38180-8732-11eb-991b-345a896a195f.mp4

### Actual behaviour

The `func_name` of the first placeholder is `1G`.

```vim
function! 1G(<`2`>) abort
        <`0:TARGET`>
endfunction
```

### Expected behaviour

The first placeholder should be `func_name`.

```vim
function! func_name(<`2`>) abort
        <`0:TARGET`>
endfunction
```

**Comment by dstein64** [comment]
Thanks for reporting this @tamago324!

The plugin was using `normal!` commands that are treated as input text when Neovim is in select-mode (e.g., `normal! 1G`). In PR #29, I've updated the plugin so that select-mode is temporarily changed to visual mode for the duration of scrollbar refreshing.

**Comment by tamago324** [comment]
It's working correctly! Thank you very much!

**Comment by YaBoiBurner** [comment]
I'm still running into this issue with vim-vsnip. Did a lot of testing with it, and it only happens on larger files now.

The effects aren't as reliable anymore either - now large garbage edits get made instead `1G` getting inserted.

Example:

https://user-images.githubusercontent.com/10004812/111811405-05030f00-88a5-11eb-99f1-cf6b778699de.mp4

I can get this to occur on https://github.com/FiloSottile/age/blob/master/x25519.go with any of the snippets provided by `gopls`. (I didn't use a recording of that because it jumped around too much to be readable.)

**Comment by dstein64** [comment]
@YaBoiBurner, thanks for reporting this!

From the video you posted, I think I know what the issue is, and I've just merged an update in PR #30 that is intended to address the problem.

I have re-opened the ticket.

Please let me know if the problem is addressed.

**Comment by YaBoiBurner** [comment]
The fix works perfectly. Thanks.
